data/campaigns/Sceptre_of_Fire/scenarios/4_Gathering_Materials.cfg

```diff
@@ -964,10 +969,10 @@
             find_vacant=no
         [/unstore_unit]
 
-        [remove_unit_overlay]
+        [remove_object]
```

Comment:
If the overhead of remove_unit_overlay is so high that it needs to be changed to remove_object, then that feels more like a bug in remove_unit_overlay than something to change in campaigns' files.

Reply:
i don't think it's a problem real problem it just feels better knowing that there are no useless objects laying around in the unit.

Reply:
If it's worth doing, I'd prefer it to be done in a way that benefits UMC too. Maybe in the unit.cpp's expire_modifications.

I won't have time until Sunday, but would be happy to do the C++ implementation on Sunday.

Reply:
> If it's worth doing, I'd prefer it to be done in a way that benefits UMC too. Maybe in the unit.cpp's expire_modifications.

i don't think its worth doing this in the c++ code, i mean i only possibility i see is to search for objects that cancel each other out but, then that'd break umc code that relied on such objects and wanted to remove the second object manually via [remove_object].. . I just don't think there is a good way. These wml changes i did were cheap and without disadvantages that's why i did them, changing it in the c++ code has the chance to break other code and for a minimal gain of reducing savefile size of a few bytes.

Reply:
I think it should:

1. Make `[remove_unit_overlay]` be no worse than `[remove_object]`.
2. Deprecate `[remove_unit_overlay]` (possibly also `[unit_overlay]`).
3. Replace all mainline uses of `[remove_unit_overlay]` with `[remove_object]`.

Reply:
If we don't deprecate the tag, then I think it should still be used in mainline, otherwise it won't be tested by playing the mainline campaigns.